docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+
+#### Create the Dev Container configuration
+1. Open the Command Palette and run the “Dev Container: Add Dev Container Configuration Files…"
+2. Select "Python 3"
+    ![Python 3 option selected in the Dev Container configuration files list.](images/fastapi-tutorial/devcontainers_python3.png)
+3. Select the default version.
+
+We can optionally install features to be included in the container. For this tutorial, we will install [Redis Server](https://github.com/itsmechlark/features/tree/main/src/redis-server), which is a community contributed feature that installs and adds the proper dev container set up for Redis.
+
+4. Select "Redis Server" as an additional feature to be installed, press "OK", and then select "Keep Defaults".
+    ![Redis Server option selected in the Dev Container configuration files list.](images/fastapi-tutorial/devcontainers_redis_server_feature.png)
+
+This will create a `.devcontainer` folder in your workspace, with a `devcontainer.json` file. Let's make some edits to this file so the container setup can include steps such as installing the VS Code extensions we'll need as well as the project dependencies.
+
+5. Open the `devcontainer.json` file.
+
+6. Locate the content below and remove the comment (`//`) from that line, so the dependencies can be installed once the container is created:
+    ```
+    "postCreateCommand": "pip3 install --user -r requirements.txt",
+    ```
+7. Add the following setting to `devcontainer.json`:
+    ```
+        // Use 'postCreateCommand' to run commands after the container is created.
+        "postCreateCommand": "pip3 install --user -r requirements.txt",
+
+        // Configure tool-specific properties.
+        "customizations": {
+            "vscode": {
+                "extensions": [
+                    "ms-python.python", //Python extension ID
+                    "ms-python.vscode-pylance" //Pylance extension ID
+                ]
+            }
+        }
+    ```
+8. Save the file.
+
+9. Select the "Reopen in Container" button from the notification that will show up on the bottom right corner.
+
+    You can learn more about dev container configuration in the [documentation](https://code.visualstudio.com/docs/devcontainers/containers#_create-a-devcontainerjson-file).
+    > Note: it may take several minutes to build the container, depending on internet speed and machine performance.
+
+Once it's done, you will have a fully configured Linux-based workspace with Python 3 and Redis Server installed.
+
+Once the container is set up, you will notice an indicator on the bottom left corner of VS Code:
+![Dev Container indicator displayed on the bottom left corner of VS Code.](images/fastapi-tutorial/devcontainer_indicator.png)
+
+You should now be ready to move on to the next section, where we will replace the data storage.
+
+## Replace the database
+1. In the `main.py` file, replace the `grocery_list = {}` in the beginning of the file with the lines below:
```

Comment:
Updating the match what we prompted the user to paste in the file most recently. 
```suggestion
1. In the `main.py` file, replace the `grocery_list: dict[int, ItemPayload] = {}` in the beginning of the file with the lines below:
```